cpp/src/merge/merge.cu

```diff
@@ -223,6 +285,56 @@ index_vector generate_merged_indices(table_view const& left_table,
   return merged_indices;
 }
 
+index_vector generate_merged_indices_nested(table_view const& left_table,
+                                            table_view const& right_table,
+                                            std::vector<order> const& column_order,
+                                            std::vector<null_order> const& null_precedence,
+                                            bool nullable,
+                                            rmm::cuda_stream_view stream)
+{
+  size_type const left_size  = left_table.num_rows();
+  size_type const right_size = right_table.num_rows();
+  size_type const total_size = left_size + right_size;
```

Comment:
Another impertinent question: Do we need to protect against the case where the `total_size` overflows?

Reply:
Good point! I added a check here https://github.com/rapidsai/cudf/pull/14250/files#diff-bea7124eddd4e287917be21bd0a0b7cb946816ad08c2a77943177a54569d87d0R641-R647